File: java/r2dbc-backend/src/main/java/chaincue/tech/r2dbcbackend/masters/course_master/CourseRepository.java
package chaincue.tech.r2dbcbackend.masters.course_master;

import org.springframework.stereotype.Component;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Component
public interface CourseRepository {
    Flux<Course> findAll();

    Mono<Course> findById(String id);

    Mono<Course> findByName(String name);

    Mono<Course> save(Course department);

    Mono<Void> delete(Course department);
}

File: kotlin/jpa-backend/src/main/kotlin/chaincue/tech/jpabackend/masters/course_master/CourseRepository.kt
package chaincue.tech.jpabackend.masters.student_master

import chaincue.tech.jpabackend.masters.teacher_master.Course
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface CourseRepository : JpaRepository<Course, String>
